config/initializers/statsd.rb

```diff
@@ -33,7 +33,20 @@ class << Samson
 end
 
 ActiveSupport::Notifications.subscribe("job_status.samson") do |*, payload|
-  Samson.statsd.increment "jobs.#{payload.fetch(:type)}.#{payload.fetch(:status)}"
+  tags = []
+  if payload[:deploy]
+    tags.push("project:#{payload[:deploy].project.name}", "stage:#{payload[:deploy].stage.name}")
+    if ['succeeded', 'failed', 'errored'].include?(payload[:status])
+      total_lead_time = payload[:deploy].total_lead_time
+      if payload[:deploy].production
+        to_production_latency = payload[:deploy].to_production_latency
+        Samson.statsd.histogram "jobs.deploy.to_production_latency", to_production_latency, tags: tags
+      end
+      Samson.statsd.histogram "jobs.deploy.total_lead_time", total_lead_time, tags: tags
+    end
+  end
```

Comment:
this whole logic is rather complicated, prefer making a new class and then putting all the logic in there